frontend/src/metabase/query_builder/components/view/View/NotebookContainer/NotebookContainer.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from "react";
+
+import Notebook from "metabase/query_builder/components/notebook/Notebook";
+import { Box } from "metabase/ui";
+
+// There must exist some transition time, no matter how short,
+// because we need to trigger the 'onTransitionEnd' in the component
+const delayBeforeNotRenderingNotebook = 10;
+
+interface NotebookContainerProps {
+  isNotebookContainerOpen: boolean;
+}
+
+export const NotebookContainer = ({
+  isNotebookContainerOpen,
+  ...props
+}: NotebookContainerProps) => {
+  const [shouldShowNotebook, setShouldShowNotebook] = useState(
+    isNotebookContainerOpen,
+  );
+
+  useEffect(() => {
+    isNotebookContainerOpen && setShouldShowNotebook(isNotebookContainerOpen);
+  }, [isNotebookContainerOpen]);
+
+  const handleTransitionEnd: React.TransitionEventHandler<HTMLDivElement> = (
```

Comment:
Please destructure the import

```suggestion
  const handleTransitionEnd: TransitionEventHandler<HTMLDivElement> = (
```

Reply:
https://github.com/metabase/metabase/pull/40316/commits/4a477c9d9028e66ce62789772c49bfddb135dc9e